dockernet/ts-tests/test/main.test.ts

```diff
@@ -77,23 +82,27 @@ beforeAll(async () => {
       broadcastPollIntervalMs: 50,
       resolveIbcResponsesCheckIntervalMs: 50,
     });
-  }
 
-  console.log("waiting for chain to start...");
-  while (true) {
-    try {
-      const block =
-        await accounts.user.query.cosmos.base.tendermint.v1beta1.getLatestBlock(
-          {},
-        );
+    if (name === "user") {
+      const signer = await Secp256k1HdWallet.fromMnemonic(mnemonic);
 
-      if (block?.block?.header.height! > 0n) {
-        break;
-      }
+      // get signer address
+      const [{ address }] = await signer.getAccounts();
 
-      await sleep(50);
-    } catch (error) {}
+      gaiaAccounts = {
+        user: await StrideClient.create(HUB_RPC_ENDPOINT, signer, address, {
+          gasPrice: GasPrice.fromString("0.025uatom"),
+          broadcastPollIntervalMs: 50,
+          resolveIbcResponsesCheckIntervalMs: 50,
+        }),
+      };
+    }
   }
+  console.log("waiting for stride to start...");
+  await waitForChain(accounts.user, "ustrd");
+
+  console.log("waiting for gaia to start...");
+  await waitForChain(gaiaAccounts.user, "uatom");
```

Comment:
just a refactor of the "waitForChain" logic

Reply:
were you running into any issues with the approach that checked for the latest block? Seems a bit more elegant that way rather than submitting a tx